### PR TITLE
Add secrets implementation for Vault Transit

### PR DESCRIPTION
**Pull request by saheienko**
Signed-off-by: Serhii Aheienko <serhii.aheienko@gmail.com>

**What this PR does / why we need it**:
Add support for the vault transit secret engine - https://www.vaultproject.io/docs/secrets/transit.

### CODE REVIEW

File: vaulttransit/vaulttransit.go [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/command/agent/auth"
+	"github.com/hashicorp/vault/command/agent/auth/kubernetes"
+	"github.com/libopenstorage/secrets"
+	"github.com/libopenstorage/secrets/pkg/store"
+	"github.com/libopenstorage/secrets/vaulttransit/client/transit"
+	"github.com/portworx/kvdb"
+)
+
+const (
```
[the remark this answers is not in the document]
Comment (saheienko) [reply vaulttransit/vaulttransit.go:20]: Moving these to the vault/utils package.

File: vaulttransit/vaulttransit.go [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+		Name:      secretID,
+		Namespace: namespace,
+	}
+}
+
+func (v *vaultSecrets) persistentStorePath(key transit.SecretKey) string {
+	return path.Join("vault", key.Namespace, key.Name)
+}
+
+func getVaultParam(secretConfig map[string]interface{}, name string) string {
```
[the remark this answers is not in the document]
Comment (saheienko) [reply vaulttransit/vaulttransit.go]: Moving to the vault/utils package.

File: vaulttransit/vaulttransit.go [hunk 3]
```diff
@@ -0,0 +1,580 @@
...
+	}
+}
+
+func New(
+	secretConfig map[string]interface{},
+) (secrets.Secrets, error) {
+	var (
+		kv kvdb.Kvdb
+	)
+	v, ok := secretConfig[VaultTransitKvdbKey]
```
[the remark this answers is not in the document]
Comment (saheienko) [reply vaulttransit/vaulttransit.go]: tls and k8s auth related stuff have been moved to the vault/utils package.

**Review by adityadani — APPROVED** [review]
lgtm! 
Thanks @saheienko for doing this!

**Pull request merged**